### PR TITLE
Add some helm chart unit tests and fix spark service account render failure when extra annotations are specified

### LINKED ISSUE #1966: helm chart failed to render spark service account when extra annotations are specified
When extra spark serviceaccount annotations  are specified in `values.yaml`:

```yaml
serviceAccounts:
  spark:
    annotations: 
      key1: value1
      key2: value2
```

Helm will failed to render template:

```shell
$ helm template spark-operator charts/spark-operator-chart -f values.yaml
Error: YAML parse error on spark-operator/templates/spark-serviceaccount.yaml: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal object into Go struct field .metadata.annotations of type string
```

### PR DESCRIPTION
**Pull request by ChenYi015**
- Add some helm chart unit tests
- Close #1966 
- Update developer guide about how to setup git pre-commit hooks and developing with helm chart

### CODE REVIEW

**Comment by vara-bonthu** [comment]
@ChenYi015, could you provide further details on the execution process of these unit tests? Also, could you verify whether you've conducted tests on these modifications using the Helm chart?

**Comment by ChenYi015** [comment]
> @ChenYi015, could you provide further details on the execution process of these unit tests? Also, could you verify whether you've conducted tests on these modifications using the Helm chart?

Detailed information about how to run helm chart unit tests can be found here [helm-unittest/helm-unittest](https://github.com/helm-unittest/helm-unittest). First, you need to install helm unit test plugin:

```shell
helm plugin install https://github.com/helm-unittest/helm-unittest.git
```

Then, run helm chart unitt tests as follows:

```shell
$ helm unittest charts/spark-operator-chart --strict

### Chart [ spark-operator ] charts/spark-operator-chart

 PASS  Test spark operator deployment   charts/spark-operator-chart/tests/deployment_test.yaml
 PASS  Test spark operator rbac charts/spark-operator-chart/tests/rbac_test.yaml
 PASS  Test spark operator service account      charts/spark-operator-chart/tests/serviceaccount_test.yaml
 PASS  Test spark rbac  charts/spark-operator-chart/tests/spark-rbac_test.yaml
 PASS  Test spark service account       charts/spark-operator-chart/tests/spark-serviceaccount_test.yaml
 PASS  Test spark operator webhook service      charts/spark-operator-chart/tests/webhook-service_test.yaml

Charts:      1 passed, 1 total
Test Suites: 6 passed, 6 total
Tests:       46 passed, 46 total
Snapshot:    0 passed, 0 total
Time:        150.450791ms
```

It will search for test files under `charts/spark-operator-chart/tests` directory with suffix `_test.yaml` and run the test suites.

**Comment by ChenYi015** [comment]
I have conducted tests on the modifications with my own k8s cluster, and it works as expected.

**Comment by vara-bonthu** [comment]
Thanks @ChenYi015 !  It would be nice to add this to our contributors.md doc

**Review by vara-bonthu — APPROVED** [review]
/approve

**Comment by ChenYi015** [comment]
@vara-bonthu I have updated the developer guide about how to setup git pre-commit hooks and developing with helm chart.

**Review by vara-bonthu — APPROVED** [review]

**Review by vara-bonthu — COMMENTED** [review]
@ChenYi015 one more change required

File: docs/developer-guide.md [hunk 1]
```diff
@@ -72,3 +95,82 @@ make build-api-docs
 ```
 
 Running the above command will update the file `docs/api-docs.md`.
+
+## Develope with the Helm Chart
```
[the remark this answers is not in the document]
Comment (ChenYi015) [reply docs/developer-guide.md]: updated

**Review by vara-bonthu — APPROVED** [review]
/lgtm

**Pull request merged**